src/components/ViewPackagePage/components/markdown-viewer.tsx

```diff
@@ -1,15 +1,39 @@
-import React, { useEffect } from "react"
+import React, { useEffect, useState } from "react"
 import Markdown from "react-markdown"
 import remarkGfm from "remark-gfm"
 
 import { useShikiHighlighter } from "@/hooks/use-shiki-highlighter"
+import { Check, Clipboard, Copy, CopyPlus } from "lucide-react"
+import { useToast } from "@/hooks/use-toast"
 
 export default function MarkdownViewer({
   markdownContent,
 }: {
   markdownContent: string
 }) {
   const { highlighter } = useShikiHighlighter()
+  const [copiedCode, setCopiedCode] = useState<string | null>(null)
+  const { toast } = useToast()
+
+  const handleCopy = async (text: string) => {
+    try {
+      await navigator.clipboard.writeText(text)
+      setCopiedCode(text)
+      toast({
+        title: "Copied to clipboard",
+        description: "Your code has been copied successfully.",
+      })
+      setTimeout(() => {
+        setCopiedCode(null)
+      }, 1500)
```

Comment:
I don't think this is a correct practice to use `setTimeout` in this case for animation! Maybe create a custom hook or something else, cause this doesn't make it clear and took me some time to figure out the reason for this timeout, which is used for animation.

cc @seveibar

Reply:
I do see this pattern for copying sometimes, its ok but a comment would help i think eg

// changes check icon back to clipboard